Tests/L2Tests/L2TestsPlugin/tests/UsbAccess_L2Test.cpp

```diff
@@ -2,11 +2,621 @@
+
+
+    std::string directory1 = "/run/media/sda2/logs/";
+    std::vector<std::string> fileNames1 = {"file1.txt", "file2.txt", "FILE3.txt", "FILE4.png"};
+    for (const auto& fileName1: fileNames1){
+    std::string filePath1 = directory1 + fileName1;
+    std::ofstream fileStream(filePath1);
+    }
+
+    EXPECT_CALL(*p_udevImplMock, udev_list_entry_get_name(testing::_))
+         .Times(::testing::AnyNumber())
+         .WillRepeatedly(testing::Return("/dev/sda2"));
+         
+    EXPECT_CALL(*p_udevImplMock, udev_device_get_parent_with_subsystem_devtype(testing::_, testing::_, testing::_))
+        .Times(::testing::AnyNumber())
+        .WillRepeatedly(testing::Return(reinterpret_cast<struct udev_device*>(0x5)));
+        
+    EXPECT_CALL(*p_udevImplMock, udev_device_get_devtype(testing::_))
+        .Times(::testing::AnyNumber())
+        .WillRepeatedly(testing::Return("disk"));
+        
+    EXPECT_CALL(*p_udevImplMock, udev_device_get_devnode(testing::_))
+        .Times(::testing::AnyNumber())
+        .WillRepeatedly(testing::Return("/dev/sda2"));
+
+    EXPECT_CALL(*p_wrapsImplMock, getmntent(testing::_))
+        .Times(::testing::AnyNumber())
+        .WillRepeatedly(::testing::Invoke(
+        [&](FILE*) -> struct mntent* {
+        static struct mntent entry1;
+        static struct mntent entry2;
+
+        entry1.mnt_fsname = const_cast<char*>("/dev/sda1"); // Set the value for mnt_fsname
+        entry1.mnt_dir = const_cast<char*>("/run/media/sda1/logs");    // Set the value for mnt_dir
+
+        entry2.mnt_fsname = const_cast<char*>("/dev/sda2"); // Set the value for mnt_fsname
+        entry2.mnt_dir = const_cast<char*>("/run/media/sda2/logs/");    // Set the value for mnt_dir
+
+        static int callCount = 1;
+        if (callCount == 0) {
+            callCount++;
+            return &entry1;
+        } else if (callCount == 1) {
+            callCount++;
+            return &entry2;
+        } else {
+            return static_cast<struct mntent*>(NULL);
+        }
+    }));
+
+
+    status = InvokeServiceMethod("org.rdk.UsbAccess.1", "getFileList", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    message = "{\"mounted\":true,\"device\":\"\\/dev\\/sda2\"}";
+    expected_status.FromString(message);
+    EXPECT_CALL(async_handler, onUSBMountChanged(MatchRequestStatus(expected_status)))
+        .WillOnce(Invoke(this, &UsbAccess_L2test::onUSBMountChanged));
+       
+    IARM_Bus_SYSMgr_EventData_t usbEventData1;
+    usbEventData1.data.usbMountData.mounted= true;
+    strncpy(usbEventData1.data.usbMountData.dir, "/dev/sda2", sizeof(usbEventData1.data.usbMountData.dir));
+    iarmEventHandler(IARM_BUS_SYSMGR_NAME, IARM_BUS_SYSMGR_EVENT_USB_MOUNT_CHANGED, &usbEventData1, 1);
+ 
+    signalled = WaitForRequestStatus(JSON_TIMEOUT,UsbAccess_onUSBMountChanged);
+    EXPECT_TRUE(signalled & UsbAccess_onUSBMountChanged);
+
```

Comment:
I don't see anywhere the getFileList response is compared. This api should respond with the file lists in the given path and the testcase should compare those filelists and determine pass/fail.